### PR TITLE
✅ [amp story player] Visual Tests for Back and Close Button

### PR DESCRIPTION
**Pull request by connielei**
- Add visual diff tests for player exit controls

Requires #29372

### CODE REVIEW

File: src/amp-story-player/amp-story-player-impl.js [hunk 1]
```diff
@@ -221,14 +221,18 @@ export class AmpStoryPlayer {
   initializeShadowRoot_() {
     this.rootEl_ = this.doc_.createElement('main');
 
-    // Create shadow root
-    const shadowRoot = this.element_.attachShadow({mode: 'open'});
+    // TODO(): Update unit tests to work without shadow root,
+    // then update condition to getMode.test and update visual tests.
+    const containerToUse =
+      typeof __AMP_VISUAL_TEST !== 'undefined'
```
Comment (danielrozenberg) [inline src/amp-story-player/amp-story-player-impl.js:227]: I believe this feature already exists as `__AMP_TEST`, but either @choumx or @newmuis will know better than me :)

File: examples/visual-tests/amp-story-player/back-button.html [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<html>
+  <head>
+    <title>Story Player (Non-AMP)</title>
+    <script>
+      // TODO(): Remove this when unit tests for player are fixed.
+      __AMP_VISUAL_TEST = true;
+    </script>
+    <script async src="../../../dist/amp-story-player.js"></script>
+    <link href="../../../build/css/amp-story-player-v0.css" rel='stylesheet' type='text/css'>
```
Comment (danielrozenberg) [inline examples/visual-tests/amp-story-player/back-button.html:9]: Not sure if this is related, but try to include these files like you would in a normal AMP page (`https://cdn.ampproject.org/amp-story-player-v0.js` and `.css`) - the internal server should rewrite them correctly both locally and on Travis. Try this and let me know if this makes things better :)

Comment (connielei) [reply examples/visual-tests/amp-story-player/back-button.html:9]: Tried, did not work. [Build for change urls commit](https://percy.io/ampproject/amphtml/builds/6149656)

File: src/amp-story-player/amp-story-player-impl.js (hunk 1 above)
Comment (newmuis) [reply src/amp-story-player/amp-story-player-impl.js:227]: Yep, it's a bug on the stories side; the player unit tests currently (incorrectly) *require* shadow DOM to be enabled, so we need to disable shadow DOM for visual tests but not unit tests.

We should file a GitHub issue to do that work and include the issue number here (e.g. `TODO(#12345): ...`)

Comment (gmajoulet) [reply src/amp-story-player/amp-story-player-impl.js]: I think using this helper instead should work: https://github.com/ampproject/amphtml/blob/5059ece5035547b50530714fe443ba41c9223b69/extensions/amp-story/1.0/utils.js#L95-L112

**Comment by Enriqe** [comment]
I have a PR out https://github.com/ampproject/amphtml/pull/29406 where I was able to run the visual/interactive tests for the player. Feel free to take a look and grab whatever is useful from there (there's a bunch of infra changes to make the debugging easier, feel free to ignore those).

Thanks again to @estherkim and @danielrozenberg for their valuable help on this!!

File: examples/visual-tests/amp-story-player/back-button.html (hunk 2 above)
Comment (connielei) [reply examples/visual-tests/amp-story-player/back-button.html]: Error was script should have been the minified version.

**Comment by connielei** [comment]
> I have a PR out #29406 where I was able to run the visual/interactive tests for the player. Feel free to take a look and grab whatever is useful from there (there's a bunch of infra changes to make the debugging easier, feel free to ignore those).
> 
> Thanks again to @estherkim and @danielrozenberg for their valuable help on this!!

And @caroqliu!

File: examples/visual-tests/amp-story-player/placeholder.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+
+
```
Comment (Enriqe) [inline examples/visual-tests/amp-story-player/placeholder.js:2]:
```suggestion
```

File: examples/visual-tests/amp-story-player/back-button.html [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      /* TODO: Remove this when poster image is no longer displayed. */
+      amp-story-player a:first-of-type {
+        display: none;
+      }
+    </style>
+    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+  </head>
+  <body>
+    <amp-story-player style="width: 360px; height: 600px;" exit-control="back-button">
+      <a href="https://www-washingtonpost-com.cdn.ampproject.org/v/s/www.washingtonpost.com/graphics/2019/lifestyle/travel/amp-stories/a-locals-guide-to-what-to-eat-and-do-in-new-york-city/"
```
Comment (Enriqe) [inline examples/visual-tests/amp-story-player/back-button.html:48]: Can we use a local story instead?

File: examples/visual-tests/amp-story-player/placeholder.js (hunk 3 above)
Comment (connielei) [reply examples/visual-tests/amp-story-player/placeholder.js]: I did not notice that! Fixed.

**Pull request merged**